mbuild/lattice.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+from collections import defaultdict
+
+__all__ = ['Lattice']
+
+
+class Lattice(object):
+    """A building block for regular, homogeneous or heterogeneous materials.
+
+    Lattice is the abstract building block of any object that exhibits
+    long-range order. More specfically, any object that has particles arranged
+    in a predictable, infinitely repeating pattern with the particles
+    occupying the cell.
+    These lattice sites are related to the origin by a set of translation
+    vectors.
+
+    The long range lattice can be reduced to a smaller, more manageable "unit
+    cell". In other words, a smaller unit of the lattice, when repeated
+    infinitely, will generate the long range lattice.
+
+    TODO: Add in orientation support similar to lammps implementation possibly
+
+    Parameters
+    ----------
+    dimension : integer, optional, default 3
+        Dimension of the lattice, can be 2 or 3
+    lattice_vectors : "SOME ARRAY", optional, default [1,0,0] [0,1,0] [0,0,1]
```

Comment:
Try to stay consistent with the array formatting for docstrings. E.g.:
https://github.com/mdtraj/mdtraj/blob/master/mdtraj/geometry/rdf.py#L41